radixdlt/src/main/java/com/radixdlt/consensus/liveness/PacemakerImpl.java

```diff
@@ -98,14 +98,15 @@ public View getCurrentView() {
 			return Optional.empty();
 		} else {
 			// if we got enough new-views, remove pending and return formed QC
-			pendingNewViews.remove(newView.getView());
 
-			if (newView.getView().compareTo(this.currentView) > 0) {
+			if (newView.getView().compareTo(this.currentView) > 0
+				&& newView.getQC().getView().equals(this.currentView)) {
 				this.currentView = newView.getView();
 				scheduleTimeout(this.currentView);
 			}
 
-			if (newView.getView().compareTo(this.currentView) >= 0) {
+			if (newView.getView().equals(this.currentView)) {
+				pendingNewViews.remove(newView.getView());
```

Comment:
Shouldn't the entry in `pendingNewViews` be removed either way when a quorum is formed, regardless of whether it represents a new view?

Reply:
no, should still wait for previous leader's new-view. If we removed, then we would need to wait for the next timeout which we don't want.

Reply:
In that case comment at line 100 should be updated